### PR TITLE
Wireup ibc message page

### LINKED ISSUE #1138: Wireup IBC message page

### PR DESCRIPTION
**Pull request by prin-r**
Fix #1138

### CODE REVIEW

**Comment by prin-r** [comment]
This PR is not complete yet.

**Comment by evilpeach** [comment]
@sorawit Should #1246 merge before this ticket?

File: scan/src/subscriptions/IBCSub.re [hunk 1]
```diff
@@ -38,57 +40,130 @@ type t = {
...
-  {
-    direction: Outgoing,
-    chainID: "Mumu network 01",
-    chennel: "azjlvvthyj",
-    port: "mumian_port",
-    blockHeight: ID.Block.ID(20999),
+type internal_t = {
+  isIncoming: bool,
+  blockHeight: ID.Block.t,
+  chennel: string,
```
Comment (sorawit) [inline scan/src/subscriptions/IBCSub.re]: typo?

File: scan/src/subscriptions/IBCSub.re [hunk 2]
```diff
@@ -38,57 +40,130 @@ type t = {
...
-        result: Some("aa" |> JsBuffer.fromHex),
-      }),
-  },
-  {
-    direction: Outgoing,
-    chainID: "Mumu network 01",
-    chennel: "azjlvvthyj",
-    port: "mumian_port",
-    blockHeight: ID.Block.ID(20999),
+type internal_t = {
```
Comment (sorawit) [inline scan/src/subscriptions/IBCSub.re]: Let's wrap internal type in its own module.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/subscriptions/IBCSub.re [hunk 3]
```diff
@@ -36,59 +38,134 @@ type packet_t =
...
+                : None,
+          },
+        );
+      },
+  };
+
+  module MultiPacketsConfig = [%graphql
+    {|
+    subscription Requests($limit: Int!, $offset: Int!) {
+      packets(limit: $limit, offset: $offset) @bsRecord {
```
Comment (evilpeach) [inline scan/src/subscriptions/IBCSub.re]: add orderBy block height (DESC)

File: scan/src/subscriptions/IBCSub.re [hunk 4]
```diff
@@ -36,59 +38,134 @@ type packet_t =
...
+                ? Some(packetDetail |> at(["value", "result"], string) |> JsBuffer.fromHex)
+                : None,
+          },
+        );
+      },
+  };
+
+  module MultiPacketsConfig = [%graphql
+    {|
+    subscription Requests($limit: Int!, $offset: Int!) {
```
Comment (evilpeach) [inline scan/src/subscriptions/IBCSub.re]: Rename Requests to Packets

File: scan/src/components/Packet.re [hunk 5]
```diff
@@ -36,25 +46,38 @@ let make = (~packet: IBCSub.packet_t) => {
...
-          [KVTable.Value("crypto_symbol"), KVTable.Value("BTC")],
-          [KVTable.Value("aggregation_method"), KVTable.Value("mean")],
-          [
-            KVTable.Value("data_sources"),
-            KVTable.Value("Binance v1, coingecko v1, coinmarketcap v1, band-validator"),
-          ],
-        ]
-      />
-      <VSpacing size=Spacing.xl />
+      {let calldataRender =
```
Comment (evilpeach) [inline scan/src/components/Packet.re]: Let's name calldataHeadRender

File: scan/src/components/Packet.re [hunk 6]
```diff
@@ -114,24 +145,46 @@ let make = (~packet: IBCSub.packet_t) => {
...
-           <VSpacing size=Spacing.md />
-           // TODO: Mock calldata
-           <KVTable
-             tableWidth=470
-             rows=[
-               [KVTable.Value("px"), KVTable.Value("70000")],
-               [KVTable.Value("timestamp"), KVTable.Value("1586319310")],
-             ]
-           />
+           {let resultRender =
```
Comment (evilpeach) [inline scan/src/components/Packet.re]: Let's name `resultHeaderRender`

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/subscriptions/IBCSub.re [hunk 7]
```diff
@@ -36,59 +36,128 @@ type packet_t =
...
-        oracleScriptID: ID.OracleScript.ID(20777),
-        oracleScriptName: "Mock Oracle Script",
-        status: Response.Fail,
-        result: None,
-      }),
-  },
-|];
+      switch (packetType) {
+      | "ORACLE REQUEST" =>
+        Js.Console.log(packetType);
```
Comment (evilpeach) [inline scan/src/subscriptions/IBCSub.re]: remove log

File: scan/src/components/Packet.re [hunk 8]
```diff
@@ -114,24 +136,46 @@ let make = (~packet: IBCSub.packet_t) => {
...
+           | _ => None
+           };
          <>
            <VSpacing size=Spacing.lg />
-           <div className=Styles.hFlex>
-             <Text value="CALLDATA" size=Text.Sm weight=Text.Thin spacing={Text.Em(0.06)} />
-             <HSpacing size=Spacing.md />
-             <CopyButton data=result />
-           </div>
+           {let resultHeaderRender =
```
Comment (evilpeach) [inline scan/src/components/Packet.re]: `resultHeadRender`

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/components/Packet.re [hunk 9]
```diff
@@ -114,24 +138,48 @@ let make = (~packet: IBCSub.packet_t) => {
       </div>
       {switch (response.status, response.result) {
        | (IBCSub.Response.Success, Some(result)) =>
+         let outputKVsOpt =
```
Comment (evilpeach) [inline scan/src/components/Packet.re:143]: add TODO: support loading state, no data later

**Review by evilpeach — DISMISSED** [review]
LGTM

File: scan/src/components/Packet.re [hunk 10]
```diff
@@ -15,12 +15,22 @@ module Styles = {
...
 
 [@react.component]
-let make = (~packet: IBCSub.packet_t) => {
+let make = (~packet: IBCSub.packet_t, ~oracleScriptID: ID.OracleScript.t) => {
+  // TODO: If we can get the schema out of IBCSub directly then this sub is not necessary any more.
+  let schemaSub = OracleScriptCodeSub.getSchemaByOracleScriptID(oracleScriptID);
+
   switch (packet) {
   | IBCSub.Request(request) =>
+    let outputKVsOpt =
```
Comment (sorawit) [inline scan/src/components/Packet.re:30]: also here?

File: scan/src/subscriptions/IBCSub.re [hunk 11]
```diff
@@ -36,59 +36,127 @@ type packet_t =
...
+      switch (packetType) {
+      | "ORACLE REQUEST" =>
+        Request(
+          JsonUtils.Decode.{
+            id: ID.Request.ID(packetDetail |> at(["request_id"], int)),
+            oracleScriptID: ID.OracleScript.ID(packetDetail |> at(["oracle_script_id"], int)),
+            oracleScriptName: packetDetail |> at(["oracle_script_name"], string),
+            clientID: packetDetail |> at(["client_id"], string),
+            calldata: packetDetail |> at(["calldata"], string) |> JsBuffer.fromHex,
+            requestedValidatorCount: packetDetail |> at(["ask_count"], int),
```
Comment (sorawit) [inline scan/src/subscriptions/IBCSub.re:93]: bug?

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**